File: JE/1_22/boj_10866.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class boj_10866 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Deque<String> d = new LinkedList<>();
        int t = Integer.parseInt(br.readLine());
        while (t-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String input1 = st.nextToken();

            switch (input1) {
                case "push_front":
                    d.addFirst(st.nextToken());
                    break;
                case "push_back":
                    d.addLast(st.nextToken());
                    break;
                case "pop_front":
                    if (d.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(d.pollFirst()).append("\n");
                    break;
                case "pop_back":
                    if (d.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(d.pollLast()).append("\n");
                    break;
                case "size":
                    sb.append(d.size()).append("\n");
                    break;
                case "empty":
                    if (d.isEmpty())
                        sb.append("1").append("\n");
                    else
                        sb.append("0").append("\n");
                    break;
                case "front":
                    if (d.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(d.peekFirst()).append("\n");
                    break;
                case "back":
                    if (d.isEmpty())
                        sb.append("-1").append("\n");
                    else
                        sb.append(d.peekLast()).append("\n");
                    break;
            }
        }
        System.out.println(sb);

    }
}

File: Lim-YongKwan/1월 28일/BOJ_10866.cpp
#include<iostream>
#include<deque>
#define endl "\n"

using namespace std;

int main() {

	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n;
	cin >> n;
	deque<int> silver;

	while (n--) {
		string s;
		cin >> s;

		if (s == "push_front") {
			int temp;
			cin >> temp;
			silver.push_front(temp);
		}
		else if (s == "push_back") {
			int temp;
			cin >> temp;
			silver.push_back(temp);
		}
		else if (s == "pop_front") {
			if (silver.size() > 0) {
				cout << silver.front() << endl;
				silver.pop_front();
			}
			else {
				cout << -1 << endl;
			}
		}
		else if (s == "pop_back") {
			if (silver.size() > 0) {
				cout << silver.back() << endl;
				silver.pop_back();
			}
			else {
				cout << -1 << endl;
			}
		}
		else if (s == "size") {
			cout << silver.size() << endl;
		}
		else if (s == "empty") {
			if (silver.empty()) {
				cout << 1 << endl;
			}
			else
				cout << 0 << endl;
		}
		else if (s == "front") {
			if (silver.size() > 0) {
				cout << silver.front() << endl;
			}
			else {
				cout << -1 << endl;
			}
		}
		else if (s == "back") {
			if (silver.size() > 0) {
				cout << silver.back() << endl;
			}
			else
				cout << -1 << endl;
		}

	}

	return 0;
}